### PR TITLE
Build package

### PR DESCRIPTION
**Pull request by nicolas-enjalbert**
Build package and add the first version for IIDEA

### CODE REVIEW

**Comment by pneuvial** [comment]
Thanks!

Please update the DESCRIPTION file according to requirements.txt, some packages need to be imported from

**Comment by pneuvial** [comment]
Some of the functions in R/fct_post-hoc.R are pasted from the sanssouci package. Since the package already imports sanssouci, why not using these functions directly?  In the case of curveMaxFP, currently it is not exported but we could export it in the next release if necessary.

File: R/fct_output.R [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  categ <- colnames(data$matrix)
+  data$categ <- rep(1, length(categ))
+  data$categ[which(categ == "NEG")] <- 0
+
+  colnames(data$matrix) <- data$categ
+
+
+  # setProgress(value = 0.4, detail = "Fc and p.value matrix ... ")
+
+  #### degraded matrix
```
Comment (pneuvial) [inline R/fct_output.R:33]: vocabulary: call this "light" matrix, not degraded matrix!

File: R/fct_post-hoc.R [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+#' @param alpha numeric value in (0,1)
+#' @param k integer value in `[1,m]`
+#' @param m integer value
+#'
+#' @export
+#' @examples
+#' m <- 10
+#' alpha <- 0.05
+#' thr <- t_linear(alpha, 1:m, m)
+t_linear <- function(alpha, k, m) {
```
Comment (pneuvial) [inline R/fct_post-hoc.R]: this function is part of sanssouci

File: R/fct_post-hoc.R [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+#' @return A vector of size \eqn{s} giving an joint upper confidence bound on
+#'   the number of false discoveries among the \eqn{k} most significant items
+#'   for all \eqn{i \in \{1\ldots s\}}.
+
+#' @author Gilles Blanchard, Nicolas Enjalbert-Courrech, Pierre Neuvial and
+#'   Etienne Roquain
+#' @details The time and space complexity of this function is O(s), which is
+#'   optimal since s is the length of the returned vector.
+curveMaxFP <- function(p.values, thr) {
```
Comment (pneuvial) [inline R/fct_post-hoc.R]: this function is part of sanssouci: import from it

File: R/fct_post-hoc.R (hunk 2 above)
Comment (pneuvial) [inline R/fct_post-hoc.R]: this function is part of sanssouci: import from it

File: ShinyApps/ui.R [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                       uiOutput("errorBioMatrix"),
+                       plotly::plotlyOutput("volcanoplotPriori", height = "600px")
+                       # plotlyOutput(outputId = "curveMaxFPGroup")
+      )
+    )
+  ),
+  p(em("This interactive ",
+       a("shiny", href = "https://shiny.rstudio.com", target="_blank"),
+       "application is developed by",
+       "Nicolas Enjalbert-Courrech",
```
Comment (pneuvial) [inline ShinyApps/ui.R]: add link to your webpage!

File: ShinyApps/server.R [hunk 5]
```diff
@@ -0,0 +1,1373 @@
...
+    req(object_I())
+    newValue <- ifelse(input$refFamily == "Beta",
+                       round(2*req(nrow(object_I()$input$Y))/100),
+                       req(nrow(object_I()$input$Y)))
+    numKI(newValue)
+  })
+
+  ## numK is the parameters choosen by users and use in server side
+  numK <- reactiveVal()
+  observeEvent(object_I(),{   # si les paramètres ne sont pas activés, input$valueK n'existe pas et donc on a pas le bon résultat ... pourquoi ? CalibrateJER devrait gérer si K est null non ?
```
Comment (pneuvial) [inline ShinyApps/server.R]: please translate French comments into English

File: ShinyApps/server.R [hunk 6]
```diff
@@ -0,0 +1,1373 @@
...
+  })
+
+
+  ###################
+  # Calibration
+  ###################
+
+
+
+  # ## JER calibration for available expression gene matrix
```
Comment (pneuvial) [inline ShinyApps/server.R]: Please drop chunks of commented code

**Comment by pneuvial** [comment]
I've only reviewed the functions in ShinyApp for now. Hope to do the rest soon.

**Comment by pneuvial** [comment]
Thanks for the updates.
The package passes R check with 3 NOTES. I've fixed one (adding files to .Rbuildignore) but there are still issues with 

- external functions should be imported from
- calls to 'data' (not sure what is the best way to fix this)

 Can you fix these issues? We can discuss the 2nd one if needed.

**Comment by pneuvial** [comment]
Thanks. I thnik we should export curveMaxFP and formatBounds in sanssouci, so that we can use them here.

**Comment by nicolas-enjalbert** [comment]
@pneuvial I think the pull request can be merged. What are you thinking about it ? Thanks.

**Pull request merged**